src/Util/ComposerAutoloaderFinder.php

```diff
@@ -19,45 +19,87 @@
  */
 class ComposerAutoloaderFinder
 {
+    private $rootNamespace;
+
     /**
      * @var ClassLoader|null
      */
     private $classLoader = null;
 
+    public function __construct(string $rootNamespace)
+    {
+        $this->rootNamespace = [
+            'psr0' => rtrim($rootNamespace, '\\'),
+            'psr4' => rtrim($rootNamespace, '\\').'\\',
+        ];
+    }
+
     public function getClassLoader(): ClassLoader
     {
         if (null === $this->classLoader) {
-            $this->classLoader = $this->findComposerClassLoader();
+            $this->findComposerClassLoader();
         }
 
         if (null === $this->classLoader) {
-            throw new \Exception('Composer ClassLoader not found!');
+            throw new \Exception("Could not find a Composer autoloader that autoloads from '{$this->rootNamespace['psr4']}'");
         }
 
         return $this->classLoader;
     }
 
-    /**
-     * @return ClassLoader|null
-     */
     private function findComposerClassLoader()
     {
         $autoloadFunctions = spl_autoload_functions();
 
         foreach ($autoloadFunctions as $autoloader) {
-            if (\is_array($autoloader) && isset($autoloader[0]) && \is_object($autoloader[0])) {
-                if ($autoloader[0] instanceof ClassLoader) {
-                    return $autoloader[0];
-                }
-
-                if ($autoloader[0] instanceof DebugClassLoader
-                    && \is_array($autoloader[0]->getClassLoader())
-                    && $autoloader[0]->getClassLoader()[0] instanceof ClassLoader) {
-                    return $autoloader[0]->getClassLoader()[0];
-                }
+            $classLoader = $this->extractComposerClassLoader($autoloader);
+            if ($classLoader && $this->locateMatchingClassLoader($classLoader)) {
+                return;
+            }
+        }
+    }
+
+    /**
+     * @return ClassLoader|null
+     */
+    private function extractComposerClassLoader(array $autoloader)
+    {
+        if (isset($autoloader[0]) && \is_object($autoloader[0])) {
+            if ($autoloader[0] instanceof ClassLoader) {
+                return $autoloader[0];
+            }
+            if ($autoloader[0] instanceof DebugClassLoader
+                && \is_array($autoloader[0]->getClassLoader())
+                && $autoloader[0]->getClassLoader()[0] instanceof ClassLoader) {
+                return $autoloader[0]->getClassLoader()[0];
             }
         }
 
         return null;
     }
+
+    private function locateMatchingClassLoader(ClassLoader $classLoader): bool
+    {
+        foreach ($classLoader->getPrefixesPsr4() as $prefix => $paths) {
+            // We can default to using the autoloader containing this component if none are matching.
+            if ('Symfony\\Bundle\\MakerBundle\\' === $prefix) {
+                $this->classLoader = $classLoader;
+            }
```

Comment:
I'm not sure about this. Well, specifically, it makes sense. But, what edge case are we coding for? It definitely complicates the algorithm (all my thoughts about returning from these methods instead of setting the property directly fall apart).